compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -463,14 +451,28 @@ pub trait PrettyPrinter<'tcx>:
             // `visible_parent_map`), looking for the specific child we currently have and then
             // have access to the re-exported name.
             DefPathData::TypeNs(ref mut name) if Some(visible_parent) != actual_parent => {
-                let reexport = self
+                // Item might be re-exported several times
+                let reexports: Vec<_> = self
                     .tcx()
                     .item_children(visible_parent)
                     .iter()
-                    .find(|child| child.res.opt_def_id() == Some(def_id))
-                    .map(|child| child.ident.name);
-                if let Some(reexport) = reexport {
-                    *name = reexport;
+                    .filter(|child| child.res.opt_def_id() == Some(def_id))
+                    .collect();
+
+                if reexports.len() > 0 {
+                    if let Some(reexport) = reexports
+                        .into_iter()
+                        .find(|child| child.vis.is_public() && child.ident.name != kw::Underscore)
+                        .map(|child| child.ident.name)
```

Comment:
Do we actually need to collect into a vec?

Reply:
Good point. Probably not.